src/main/java/core/basesyntax/service/operation/PurchaseOperationHandler.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.service.operation;
+
+import core.basesyntax.service.FruitTransaction;
+import java.util.Map;
+
+public class PurchaseOperationHandler implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> storage) {
+        String fruit = transaction.getFruit();
+        int amount = transaction.getAmount();
+        int currentAmount = storage.getOrDefault(fruit, 0);
+
+        if (currentAmount >= amount) {
+            storage.put(fruit, currentAmount - amount);
+        } else {
+            throw new RuntimeException("Not enough fruit in storage: " + fruit);
```

Comment:
The exception message should provide more context for easier debugging. Consider including the current amount of the fruit in the storage and the amount that was attempted to be purchased.